src/cargo-fmt/main.rs

```diff
@@ -282,7 +282,8 @@ fn get_targets_recursive(
     mut targets: &mut BTreeSet<Target>,
     visited: &mut BTreeSet<String>,
 ) -> Result<(), io::Error> {
-    let metadata = get_cargo_metadata(manifest_path)?;
+    let metadata = get_cargo_metadata(manifest_path, false)?;
+    let metadata_with_deps = get_cargo_metadata(manifest_path, true)?;
```

Comment:
The [dependency struct](https://docs.rs/cargo_metadata/0.8.0/cargo_metadata/struct.Dependency.html) returned from the `cargo_metadata` crate unsurprisingly does not include info about the manifest location of the dependency. 

As such, I figured I could make another call to `cargo_metadata` to get the metadata including dependencies since that cargo metadata will include the correct manifest path for the dependency, regardless of what package name/directory name are used. That "correct" manifest path is then used below.

Reply:
Calling `cargo_metadata` twice is a bit unfortunate, but I think this is the best we can do right not. A possible approach in the future would be to refactor `cargo_metadata` so that the `cargo_metadata::Dependency` contains the manifest location if the dependency is local

Reply:
Completely agree 👍